ValveResourceFormat/KeyValues/KeyValues3.cs

```diff
@@ -0,0 +1,526 @@
+        private static void ReadHeader(char c, Parser parser)
+        {
+            parser.currentString.Append(c);
+
+            //Read until --> is encountered
+            if (c == '>' && parser.currentString.ToString().Substring(parser.currentString.Length - 3) == "-->")
+            {
+                parser.stateStack.Pop();
+                parser.stateStack.Push(State.SEEK_VALUE);
+                return;
+            }
+        }
+
+        //Seeking value state
+        private static void SeekValue(char c, Parser parser)
+        {
+            //Ignore whitespace
+            if (Char.IsWhiteSpace(c) || c == '=')
+            {
+                return;
+            }
+
+            //Check struct opening
+            if (c == '{')
+            {
+                parser.stateStack.Pop();
+                parser.stateStack.Push(State.VALUE_STRUCT);
+
+                parser.objStack.Push(new KVObject(parser.currentString.ToString()));
+            }
+            //Check for array opening
+            else if (c == '[')
+            {
+                parser.stateStack.Pop();
+                parser.stateStack.Push(State.VALUE_ARRAY);
+                parser.stateStack.Push(State.SEEK_VALUE);
+
+                parser.objStack.Push(new KVObject(parser.currentString.ToString(), true));
+            }
+            //Check for array closing
+            else if (c == ']')
+            {
+                parser.stateStack.Pop();
+                parser.stateStack.Pop();
+
+                KVObject value = parser.objStack.Pop();
+                parser.objStack.Peek().AddProperty(value.Key, new KVValue(KVType.ARRAY, value));
+            }
+            //String opening
+            else if (c == '"')
+            {
+                //Check if a multistring or single string was found
+                string next = PeekString(parser, 4);
+                if (next.Contains("\"\"\n") || next == "\"\"\r\n")
+                {
+                    //Skip the next two "'s
+                    SkipChars(parser, 2);
+
+                    parser.stateStack.Pop();
+                    parser.stateStack.Push(State.VALUE_STRING_MULTI);
+                    parser.currentString.Clear();
+                }
+                else
+                {
+                    parser.stateStack.Pop();
+                    parser.stateStack.Push(State.VALUE_STRING);
+                    parser.currentString.Clear();
+                }
+            }
+            //Boolean false
+            else if (c == 'f' && PeekString(parser, 4) == "alse")
+            {
+                parser.stateStack.Pop();
+                
+                //Can directly be added
+                parser.objStack.Peek().AddProperty(parser.currentName, new KVValue(KVType.BOOLEAN, false));
+
+                //Skip next characters
+                SkipChars(parser, 4);
+            }
+            //Boolean true
+            else if (c == 'f' && PeekString(parser, 3) == "rue")
```

Comment:
`frue`? Looks like a copy/paste bug. Perhaps write a function that combines the current char with your read-ahead, so that you can just say `if (ReadAheadMatches("true")) { ... }`